File: MobileTracker/app/src/main/java/me/myds/g2u/mobiletracker/IconRPC/Block.java
package me.myds.g2u.mobiletracker.IconRPC;

import android.util.Log;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class Block {
    public JSONObject json_data;

    public Block(JSONObject json_data) {
        this.json_data = json_data;
    }
    public Block(String json_data) {
        try {
            this.json_data = new JSONObject(json_data);
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public String getBlockHash() {
        String block_hash = null;
        try {
            block_hash = this.json_data.getString("block_hash");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return block_hash;
    }

    public Long getTimeStamp () {
        Long time_stamp = null;
        try {
            time_stamp = this.json_data.getLong("time_stamp");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return time_stamp;
    }

    public String getPrevBlockHash() {
        String prev_block_hash = null;
        try {
            prev_block_hash = this.json_data.getString("prev_block_hash");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return prev_block_hash;
    }

    public ArrayList<Transaction> getConfirmedTransactionList () {
        ArrayList<Transaction> transactions = new ArrayList<>();

        try {
            JSONArray jsonArray = json_data.getJSONArray("confirmed_transaction_list");
            for (int i = 0; jsonArray.length() > i; i++ ) {
                JSONObject jsonObject = jsonArray.getJSONObject(i);
                transactions.add(new Transaction(jsonObject));
            }

        } catch (JSONException e) {
            e.printStackTrace();
        }

        return transactions;
    }
}

File: app/src/main/java/io/yena/mobiletracker/db/Block.kt
package io.yena.mobiletracker.db

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.util.Log
import io.yena.mobiletracker.models.BlockResult
import org.json.JSONObject

@Entity
data class Block(

    var jsonrpc: String,

    var id: Int,

    var result: String,

    @PrimaryKey var block_hash: String,

    var saved: Boolean = false

) {
    constructor(): this("", 0, "", "", false)

    fun parseResult(): BlockResult {
        val jsonObj = JSONObject(result)
        val blockResult = BlockResult()

        try {
            with(jsonObj) {
                blockResult.version = getString("version")
                blockResult.prev_block_hash = getString("prev_block_hash")
                blockResult.merkle_tree_root_hash = getString("merkle_tree_root_hash")
                blockResult.time_stamp = getString("time_stamp")
                blockResult.confirmed_transaction_list = getString("confirmed_transaction_list")
                blockResult.block_hash = getString("block_hash")
                blockResult.height = getString("height")
                blockResult.peer_id = getString("peer_id")
                blockResult.signature = getString("signature")
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return blockResult
    }
}
